### PR TITLE
feat: add workflows quickstart

### PR DESCRIPTION
**Pull request by grant**
Creates a workflows quickstart (simple program for running a workflow and waiting for results).

Based on Node quickstart: https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/master/workflows/quickstart/test/quickstart.test.js

- [ ] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [x] API's need to be enabled to test (tell us)
  - workflows.googleapis.com
- [ ] Environment Variables need to be set (ask us to set them)
- [ ] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [x] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: workflows/cloud-client/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+sa.json
```
Comment (averikitsch) [inline workflows/cloud-client/.gitignore:1]: Is this file needed?

File: workflows/cloud-client/src/main/java/com/example/workflows/WorkflowsQuickstart.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+public class WorkflowsQuickstart {
+  private static final String GOOGLE_CLOUD_PROJECT = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static String LOCATION = System.getenv("LOCATION");
+  private static String WORKFLOW = System.getenv("WORKFLOW");
+
+  public static String workflowsQuickstart(String projectId, String location, String workflow) {
+    // Execute workflow
+    try (ExecutionsClient workflowExecutionsClient = ExecutionsClient.create()) {
+      WorkflowName parent = WorkflowName.of(projectId, location, workflow);
+      Execution initialExecution = Execution.newBuilder().build();
```
Comment (averikitsch) [inline workflows/cloud-client/src/main/java/com/example/workflows/WorkflowsQuickstart.java:42]: Can you add a comment here on what this is doing? It might be more understandable if you use the CreateExecutionRequest.builder

File: workflows/cloud-client/src/test/java/com/example/workflows/WorkflowsQuickstartTest.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  // https://github.com/GoogleCloudPlatform/workflows-samples/blob/main/src/myFirstWorkflow.workflows.yaml
+  private static final String WORKFLOW_SOURCE = "# Copyright 2020 Google LLC\n#\n# Licensed under the Apache License, Version 2.0 (the \"License\");\n# you may not use this file except in compliance with the License.\n# You may obtain a copy of the License at\n#\n# http://www.apache.org/licenses/LICENSE-2.0\n#\n# Unless required by applicable law or agreed to in writing, software\n# distributed under the License is distributed on an \"AS IS\" BASIS,\n# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n# See the License for the specific language governing permissions and\n# limitations under the License.\n\n# [START workflows_myfirstworkflow]\n- getCurrentTime:\n call: http.get\n args:\n url: https://us-central1-workflowsample.cloudfunctions.net/datetime\n result: currentTime\n- readWikipedia:\n call: http.get\n args:\n url: https://en.wikipedia.org/w/api.php\n query:\n action: opensearch\n search: ${currentTime.body.dayOfTheWeek}\n result: wikiResult\n- returnResult:\n return: ${wikiResult.body[1]}\n# [END workflows_myfirstworkflow]\n";
```
Comment (averikitsch) [inline workflows/cloud-client/src/test/java/com/example/workflows/WorkflowsQuickstartTest.java:38]: Is there any way to make this easy to read? I understand this is not the focus of this sample, but it would be nicer to read from a YAML

File: workflows/cloud-client/.gitignore (hunk 1 above)
Comment (grant) [reply workflows/cloud-client/.gitignore:1]: I believe a local service account is needed for local testing.

File: workflows/cloud-client/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
+# Cloud Workflows Quickstart
```
[the remark this answers is not in the document]
Comment (grant) [reply workflows/cloud-client/README.md:1]: It seems like most instructions ask for samples in this repo. I'm not sure what difference it makes. Happy to put the samples there if that's best.

File: workflows/cloud-client/src/main/java/com/example/workflows/WorkflowsQuickstart.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+          Thread.sleep(backoffDelay);
+          backoffDelay *= 2; // Double the delay to provide exponential backoff.
+        } else {
+          System.out.printf("Execution finished with state: %s%n", execution.getState().name());
+          System.out.println(execution.getResult());
+          return execution.getResult();
+        }
+      }
+      // This return is never reached.
+      return "";
```
[the remark this answers is not in the document]
Comment (grant) [reply workflows/cloud-client/src/main/java/com/example/workflows/WorkflowsQuickstart.java:69]: I've added a polling timeout, however Java is unable to understand that a return isn't needed for this logic.

File: workflows/cloud-client/src/main/java/com/example/workflows/WorkflowsQuickstart.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    } catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (grant) [reply workflows/cloud-client/src/main/java/com/example/workflows/WorkflowsQuickstart.java:70]: Done.

File: workflows/cloud-client/src/main/java/com/example/workflows/WorkflowsQuickstart.java [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+      System.out.printf("Error executing workflow: %s%n", e);
+      return "";
+    }
+  }
+
+  /**
+   * Demonstrates using the Workflows API.
+   */
+  public static void main(String... args) {
```
[the remark this answers is not in the document]
Comment (grant) [reply workflows/cloud-client/src/main/java/com/example/workflows/WorkflowsQuickstart.java:79]: Thanks. It looks like we have seen both. Might change this if we get the quickstart test working.

File: workflows/cloud-client/src/test/java/com/example/workflows/WorkflowsQuickstartTest.java (hunk 3 above)
Comment (grant) [reply workflows/cloud-client/src/test/java/com/example/workflows/WorkflowsQuickstartTest.java:38]: It's a good question. I'd rather be able to quickly update the test with a copy/paste than format this. If we run on Java 15+, we could use text blocks.

File: workflows/cloud-client/.gitignore (hunk 1 above)
Comment (averikitsch) [reply workflows/cloud-client/.gitignore:1]: I mean do we need the extra gitignore file

File: workflows/cloud-client/src/test/java/com/example/workflows/WorkflowsQuickstartTest.java (hunk 3 above)
Comment (averikitsch) [reply workflows/cloud-client/src/test/java/com/example/workflows/WorkflowsQuickstartTest.java:42]: We can leave as is, but is there another sample we can link to that shows deploying of the YAML?

File: workflows/cloud-client/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (averikitsch) [reply workflows/cloud-client/README.md:1]: We can keep it here since it seems to fall in between [java-workflows-executions](https://github.com/googleapis/java-workflow-executions/tree/master/samples/snippets) and java-workflows

File: workflows/cloud-client/src/main/java/com/example/workflows/WorkflowsQuickstart.java [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+import com.google.cloud.workflows.executions.v1.ExecutionsClient;
+import com.google.cloud.workflows.executions.v1.WorkflowName;
+
+import java.io.IOException;
+
+/**
+ * Cloud Workflows API sample. Executes a workflow and waits for results.
+ * Example usage:
+ * GOOGLE_CLOUD_PROJECT=myProject \
+ * mvn package exec:java -Dexec.mainClass='com.example.workflows.WorkflowsQuickstart'
```
Comment (averikitsch) [inline workflows/cloud-client/src/main/java/com/example/workflows/WorkflowsQuickstart.java:34]: This should go in the README

File: workflows/cloud-client/src/main/java/com/example/workflows/WorkflowsQuickstart.java [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+      long backoffDelay = 1_000; // Start wait with delay of 1,000 ms
+      final long BACKOFF_TIMEOUT = 10 * 60_000; // Time out at 10 minutes
+      System.out.println("Poll for results...");
+      while (!executionFinished) {
+        Execution execution = workflowExecutionsClient.getExecution(executionName);
+        executionFinished = execution.getState() != Execution.State.ACTIVE;
+
+        // We've passed the max
+        if (backoffDelay > BACKOFF_TIMEOUT) {
+          return "";
```
Comment (averikitsch) [inline workflows/cloud-client/src/main/java/com/example/workflows/WorkflowsQuickstart.java:65]: Perhaps throw an error here instead? @lesv would you agree?

Comment (lesv) [reply workflows/cloud-client/src/main/java/com/example/workflows/WorkflowsQuickstart.java:65]: Yes - in general, Idiomatic Java shouldn't be returning errors, unless they are common.  If it's an exceptional situation, as this one looks to be, then we throw an exception.  Effective Java 3e Ch 69/70

**Comment by averikitsch** [comment]
@grant Will you be completing this sample?

**Comment by grant** [comment]
Closing the PR for now. May re-open later.
